applications/GeoMechanicsApplication/custom_strategies/schemes/geomechanics_time_integration_scheme.hpp

```diff
@@ -0,0 +1,349 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "solving_strategies/schemes/scheme.h"
+
+namespace Kratos {
+
+template <class TSparseSpace, class TDenseSpace>
+class GeoMechanicsTimeIntegrationScheme : public Scheme<TSparseSpace, TDenseSpace> {
+public:
+    using BaseType              = Scheme<TSparseSpace,TDenseSpace>;
+    using DofsArrayType         = typename BaseType::DofsArrayType;
+    using TSystemMatrixType     = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType     = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    void GetDofList(const Element& rElement,
+                    Element::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        const bool isActive = (rElement.IsDefined(ACTIVE)) ? rElement.Is(ACTIVE) : true;
+        if (isActive)
+            rElement.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void GetDofList(const Condition& rCondition,
+                    Element::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        const bool isActive = (rCondition.IsDefined(ACTIVE)) ? rCondition.Is(ACTIVE) : true;
```

Comment:
This line is repeated many times. Would it be better to create a method instead of the line?